recipes-extended/xen/files/libxl-xenmgr-support.patch

```diff
@@ -32,147 +34,125 @@ INTERNAL DEPENDENCIES
 ################################################################################
 PATCHES
 ################################################################################
-Index: xen-4.6.6/tools/libxl/libxl.c
+Index: xen-4.9.0/tools/libxl/libxl.h
 ===================================================================
---- xen-4.6.6.orig/tools/libxl/libxl.c
-+++ xen-4.6.6/tools/libxl/libxl.c
-@@ -1142,6 +1142,13 @@ static int libxl__domain_pvcontrol(libxl
-     return libxl__domain_pvcontrol_write(gc, XBT_NULL, domid, cmd);
- }
- 
-+int libxl_hard_shutdown(libxl_ctx *ctx, uint32_t domid)
-+{
-+    int ret;
-+    ret = xc_domain_shutdown(ctx->xch, domid, SHUTDOWN_poweroff);
-+    return ret;
-+}
-+
- int libxl_domain_shutdown(libxl_ctx *ctx, uint32_t domid)
- {
-     GC_INIT(ctx);
-@@ -1546,6 +1553,8 @@ void libxl__domain_destroy(libxl__egc *e
-         dds->stubdom.ao = ao;
-         dds->stubdom.domid = stubdomid;
-         dds->stubdom.callback = stubdom_destroy_callback;
-+        libxl__qmp_stop(gc, dds->domid);
-+        libxl__qmp_quit(gc, dds->domid);
-         libxl__destroy_domid(egc, &dds->stubdom);
-     } else {
-         dds->stubdom_finished = 1;
-@@ -3396,6 +3405,7 @@ void libxl__device_nic_add(libxl__egc *e
-     flexarray_t *front;
-     flexarray_t *back;
-     libxl__device *device;
-+    libxl_uuid uuid;
-     int rc;
-     xs_transaction_t t = XBT_NULL;
-     libxl_domain_config d_config;
-@@ -3477,6 +3487,9 @@ void libxl__device_nic_add(libxl__egc *e
-     flexarray_append(front, "mac");
-     flexarray_append(front, libxl__sprintf(gc,
-                                     LIBXL_MAC_FMT, LIBXL_MAC_BYTES(nic->mac)));
-+    libxl_domid_to_uuid(CTX, &uuid, nic->backend_domid);
-+    flexarray_append(front, "backend-uuid");
-+    flexarray_append(front, libxl__sprintf(gc, "%s", libxl__uuid2string(gc, uuid)));
- 
-     if (aodev->update_json) {
-         lock = libxl__lock_domain_userdata(gc, domid);
-@@ -3525,6 +3538,8 @@ void libxl__device_nic_add(libxl__egc *e
-     aodev->action = LIBXL__DEVICE_ACTION_ADD;
-     libxl__wait_device_connection(egc, aodev);
+--- xen-4.9.0.orig/tools/libxl/libxl.h
++++ xen-4.9.0/tools/libxl/libxl.h
+@@ -1405,6 +1405,7 @@ int libxl_domain_remus_start(libxl_ctx *
+                              const libxl_asyncop_how *ao_how)
+                              LIBXL_EXTERNAL_CALLERS_ONLY;
  
-+    libxl__xs_write(gc, XBT_NULL, GCSPRINTF("%s/hotplug-status", libxl__device_backend_path(gc, device)), "connected");
```

Comment:
fix: cool, thanks.
I do have a preference for a separate PR for the edit because I use the single URL for each PR to cover IP ownership reporting and it would create more work to clarify separate contributions. Hope that's ok.